clustering/jgroups/extension/src/main/java/org/jboss/as/clustering/jgroups/subsystem/LegacyProtocolResourceDefinition.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2018, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.as.clustering.jgroups.subsystem;
+
+import java.util.function.Consumer;
+import java.util.function.UnaryOperator;
+
+import org.jboss.as.clustering.controller.Operations;
+import org.jboss.as.clustering.controller.ResourceDescriptor;
+import org.jboss.as.clustering.controller.ResourceServiceBuilderFactory;
+import org.jboss.as.clustering.jgroups.logging.JGroupsLogger;
+import org.jboss.as.controller.OperationContext;
+import org.jboss.as.controller.OperationStepHandler;
+import org.jboss.as.controller.PathAddress;
+import org.jgroups.stack.Protocol;
+import org.wildfly.clustering.jgroups.spi.ChannelFactory;
+
+/**
+ * Resource definition for legacy protocols.
+ * @author Paul Ferraro
+ */
+public class LegacyProtocolResourceDefinition<P extends Protocol> extends ProtocolResourceDefinition<P> {
+
+    private static class OperationTransformation implements UnaryOperator<OperationStepHandler> {
+        private final String targetName;
+
+        OperationTransformation(String targetName) {
+            this.targetName = targetName;
+        }
+
+        @Override
+        public OperationStepHandler apply(OperationStepHandler handler) {
+            return (context, operation) -> {
```

Comment:
This should not be a lambda.

Reply:
Is there really anything to be gained by replacing this with an anonymous class?

Reply:
According to Andrew Dinn:

"As Bill says every lambda requires its own anonymous class and that adds
about 700 /used/ bytes to metaspace. Unfortunately, the cost is more
than that. The JVM currently places Lambda classes in their own class
loader. That is done so they can be garbage collected should the method
handle at the lambda call site be updated. That means the 700 used bytes
get allocated out of a 1.5K memory chunk dedicated to that class loader.
Since the loader has only one class none of the remaining free space can
be used for other metadata. So, the real /used/ count for a lambda is
the full 1.5K."

So basically it saves one class loader and its associated metaspace allocation.

Reply:
Fixed.